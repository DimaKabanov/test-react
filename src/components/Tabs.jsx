import React, { Component } from 'react';
import 'react-tabs/style/react-tabs.css';
import {
  Tabs as TabsWrapper,
  Tab,
  TabList,
  TabPanel,
} from 'react-tabs';

class Tabs extends Component {
  renderTab = tabs => tabs.map(tab => <Tab data-tab-id={`tab-${tab.id}`} key={tab.id}>{tab.title}</Tab>)

  renderTabPanel = tabs => tabs.map(tab => <TabPanel key={tab.id}>{tab.text}</TabPanel>)

  render() {
    const { tabs } = this.props;
    return (
      <TabsWrapper>
        <TabList>
          {this.renderTab(tabs)}
        </TabList>
        <React.Fragment>
          {this.renderTabPanel(tabs)}
        </React.Fragment>
      </TabsWrapper>
    );
  }
}

export default Tabs;
