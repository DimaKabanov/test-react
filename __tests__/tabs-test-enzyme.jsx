import Tabs from '../src/components/Tabs';

const tabsData = [
  {
    id: 0,
    title: 'Mario',
    text: 'Mario ipsum dolor sit, amet consectetur adipisicing elit.',
  },
  {
    id: 1,
    title: 'Peach',
    text: 'Peach ratione laudantium inventore ullam debitis optio rerum eum modi.',
  },
  {
    id: 2,
    title: 'Yoshi',
    text: 'Yoshi animi odit officia temporibus est enim, quas at facilis officiis? Itaque!',
  },
];

test('Tabs renders correctly', () => {
  const wrapper = mount(<Tabs tabs={tabsData} />);
  expect(wrapper.render()).toMatchSnapshot();

  const tabBtn1 = wrapper.find('[data-tab-id="tab-0"] .react-tabs__tab');
  const tabBtn2 = wrapper.find('[data-tab-id="tab-1"] .react-tabs__tab');
  const tabBtn3 = wrapper.find('[data-tab-id="tab-2"] .react-tabs__tab');

  tabBtn2.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  tabBtn3.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  tabBtn1.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});
