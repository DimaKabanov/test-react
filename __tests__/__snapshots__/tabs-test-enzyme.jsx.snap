// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-tab-id="tab-0"
      id="react-tabs-0"
      role="tab"
      tabindex="0"
    >
      Mario
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-1"
      id="react-tabs-2"
      role="tab"
    >
      Peach
    </li>
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-2"
      id="react-tabs-4"
      role="tab"
    >
      Yoshi
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    Mario ipsum dolor sit, amet consectetur adipisicing elit.
  </div>
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  />
</div>
`;

exports[`Tabs renders correctly 2`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-0"
      id="react-tabs-0"
      role="tab"
    >
      Mario
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-tab-id="tab-1"
      id="react-tabs-2"
      role="tab"
      tabindex="0"
    >
      Peach
    </li>
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-2"
      id="react-tabs-4"
      role="tab"
    >
      Yoshi
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel"
    id="react-tabs-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-3"
    role="tabpanel"
  >
    Peach ratione laudantium inventore ullam debitis optio rerum eum modi.
  </div>
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  />
</div>
`;

exports[`Tabs renders correctly 3`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-0"
      id="react-tabs-0"
      role="tab"
    >
      Mario
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-1"
      id="react-tabs-2"
      role="tab"
    >
      Peach
    </li>
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-tab-id="tab-2"
      id="react-tabs-4"
      role="tab"
      tabindex="0"
    >
      Yoshi
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel"
    id="react-tabs-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-5"
    role="tabpanel"
  >
    Yoshi animi odit officia temporibus est enim, quas at facilis officiis? Itaque!
  </div>
</div>
`;

exports[`Tabs renders correctly 4`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-tab-id="tab-0"
      id="react-tabs-0"
      role="tab"
      tabindex="0"
    >
      Mario
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-1"
      id="react-tabs-2"
      role="tab"
    >
      Peach
    </li>
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-tab-id="tab-2"
      id="react-tabs-4"
      role="tab"
    >
      Yoshi
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    Mario ipsum dolor sit, amet consectetur adipisicing elit.
  </div>
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-4"
    class="react-tabs__tab-panel"
    id="react-tabs-5"
    role="tabpanel"
  />
</div>
`;
